tomviz/python/tomviz/utils.py

```diff
@@ -329,6 +333,28 @@ def mark_as_tiltseries(dataobject):
     arr.SetTuple1(0, 1)
 
 
+def set_size(dataobject, x=None, y=None, z=None):
+    axes = []
+    lengths = []
+    if x is not None:
+        axes.append(0)
+        lengths.append(x)
+    if y is not None:
+        axes.append(1)
+        lengths.append(y)
+    if z is not None:
+        axes.append(2)
+        lengths.append(z)
+
+    extent = dataobject.GetExtent()
+    spacing = list(dataobject.GetSpacing())
+    for axis, new_length in zip(axes, lengths):
+        spacing[axis] = \
+            new_length / (extent[2 * axis + 1] - extent[2 * axis] + 1)
```

Comment:
Here you are calculating the spacing from the length, why not just have the API set the voxel size? I think that is what most internal code does. If we want to set the voxel size I think this is off by one - if I say the length is 2nm, and the cube has three points it only has two voxels (from 0 to 1 and from 1 to 2 or am I missing something?)

Reply:
Sorry, we do treat these as pixels, and so they effectively have an extra half on each side when we draw the slice/volume. That means that the + 1 you used / copied is correct in the way we are defining voxels having a value.